overlord/devicestate/handlers_serial.go

```diff
@@ -331,10 +331,30 @@ func prepareSerialRequest(t *state.Task, regCtx registrationContext, privKey ass
 	resp, err := client.Do(req)
 	if err != nil {
 		if !httputil.ShouldRetryError(err) {
-			// a non temporary net error, like a DNS no
-			// host, error out and do full retries
+			// a non temporary net error fully errors out and triggers a retry
+			// retries
 			return "", fmt.Errorf("cannot retrieve request-id for making a request for a serial: %v", err)
 		}
+		if httputil.NoNetwork(err) {
+			// If there is no network there is no need to count
+			// this as a tentatives attempt. If we do it this
+			// way the risk is that we tried a bunch of times
+			// with no network and if we hit the server for real
+			// and it replies with something we need to retry
+			// we will not because nTentatives is way over the
+			// limit.
+			nTentatives--
```

Comment:
Shouldn't we reset pre-poll-tenantives to 0?

Reply:
yes, that seems easier to reason about